### PR TITLE
debugpb: support fail point

### PR DESCRIPTION
**Pull request by overvenus**
Fail point API for injecting failures!

### CODE REVIEW

**Comment by javaforfun** [comment]
LGTM

File: proto/debugpb.proto [hunk 1]
```diff
@@ -99,3 +103,22 @@ message ScanMvccResponse {
     bytes key = 1;
     kvrpcpb.MvccInfo info = 2;
 }
+
+message Failure {
+    enum Type {
```
Comment (overvenus) [inline proto/debugpb.proto]: Maybe we should figure out a better name.

File: proto/debugpb.proto [hunk 2]
```diff
@@ -99,3 +103,22 @@ message ScanMvccResponse {
...
+        INVALID = 0;
+        INJECT = 1;
```
Comment (overvenus) [inline proto/debugpb.proto]: Should we remove `INVALID` and set `INJECT` as default?

File: proto/debugpb.proto [hunk 3]
```diff
@@ -99,3 +103,22 @@ message ScanMvccResponse {
...
+        RECOVER = 2;
+    }
+
+    string name = 1;
+    Type type = 2;
+    string actions = 3;
+}
+
+message FailPointRequest {
+    repeated Failure failures = 1;
```
Comment (overvenus) [inline proto/debugpb.proto]: Should it be one request for one failure?

Let's consider failures `[A, B, C]`, we may set A and B successfully but C failed, so what should we do? Revert A and B then return error to client? But it is possible that A or B is already be triggered.

**Comment by siddontang** [comment]
PTAL @BusyJay

File: proto/debugpb.proto [hunk 4]
```diff
@@ -38,6 +38,10 @@ service Debug {
     // Note: DO NOT CALL IT IN PRODUCTION, it's really expensive.
     //       Server uses keys directly w/o any encoding.
     rpc ScanMvcc(ScanMvccRequest) returns (stream ScanMvccResponse) {}
+
+    // Inject fail points. Currently, it only used in tests.
```
Comment (BusyJay) [inline proto/debugpb.proto]: s/it/it's/

File: proto/debugpb.proto (hunk 3 above)
Comment (BusyJay) [reply proto/debugpb.proto]: Just one should be OK.

File: proto/debugpb.proto (hunk 1 above)
Comment (BusyJay) [reply proto/debugpb.proto]: Add two methods instead, one for setting and one for removal.

**Comment by overvenus** [comment]
PTAL

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM
PTAL @BusyJay

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**